bundles/http/Citrix/Cherwell/Cherwell.js

```diff
@@ -0,0 +1,1278 @@
+                    },
+                    {
+                        "name": "IncidentType",
+                        "type": "STRING",
+                        "length": 32
+                    }
+                ]
+            },
+            {
+                "name": "categoriesIncidentTypes",
+                "columns": [
+                    {
+                        "name": "id",
+                        "type": "LONG",
+                        "primaryKey": true
+                    },
+                    {
+                        "name": "Category",
+                        "type": "STRING",
+                        "length": 64
+                    },
+                    {
+                        "name": "IncidentType",
+                        "type": "STRING",
+                        "length": 64
+                    }
+                ]
+            },
+            {
+                "name": "sources",
+                "columns": [
+                    {
+                        "name": "RecID",
+                        "type": "STRING",
+                        "length": 45,
+                        "primaryKey": true
+                    },
+                    {
+                        "name": "Source",
+                        "type": "STRING",
+                        "length": 64
+                    }
+                ]
+            },
+            {
+                "name": "priorities",
+                "columns": [
+                    {
+                        "name": "RecID",
+                        "type": "STRING",
+                        "length": 45,
+                        "primaryKey": true
+                    },
+                    {
+                        "name": "Priority",
+                        "type": "INTEGER"
+                    }
+                ]
+            }
+        ]
+    }
+});
+
+const pageSize = 200;
+
+async function getRequest({ dataStore, client }, moduleName, body, endpoint) {
+    let requestOptions = {
+        method: 'POST',
+        body: JSON.stringify(body)
+    };
+
+    let resp = await client.fetch(endpoint, requestOptions);
+    let moreRecords = true;
+    if (resp.ok && resp.status === 200) {
+        let data = await resp.json();
+        let dataArray = [];
+        let types = [];
+        for (let i = 0, j = 0; i < data.records.length; i++) {
+            dataArray[i] = JSON.parse(data.records[i]);
+            dataArray[i].CreatedDateTime = new Date(dataArray[i].CreatedDateTime);
+            dataArray[i].LastModifiedDateTime = new Date(dataArray[i].LastModifiedDateTime);
+            dataArray[i].LastModDateTime = new Date(dataArray[i].LastModDateTime);
+            dataArray[i].SLAResponseWarning = new Date(dataArray[i].SLAResponseWarning);
+            dataArray[i].SLAResolutionWarning = new Date(dataArray[i].SLAResolutionWarning);
+
+            if (moduleName == 'servicesIncidentTypes' || moduleName == 'categoriesIncidentTypes') {
```

Comment:
this could be simplified, 1) refactor `for` to `map` 2) walk through the result and collect types. Use `Array` operations like `filter`, `map` or Lodash functions for array manipulations, removing duplicates `_.uniqueBy` etc

Reply:
Refactored getRequest with map and uniqBy